lib/service/desktop.go

```diff
@@ -248,7 +250,29 @@ func (process *TeleportProcess) initWindowsDesktopServiceRegistered(log *logrus.
 			log.Infof("Starting Windows desktop service on %v.", listener.Addr())
 		}
 		process.BroadcastEvent(Event{Name: WindowsDesktopReady, Payload: nil})
-		err := srv.Serve(listener)
+
+		muxCAGetter := func(ctx context.Context, id types.CertAuthID, loadKeys bool) (types.CertAuthority, error) {
+			return accessPoint.GetCertAuthority(ctx, id, loadKeys)
+		}
+
+		mux, err := multiplexer.New(multiplexer.Config{
+			Listener:                    listener,
+			EnableExternalProxyProtocol: cfg.Proxy.EnableProxyProtocol,
+			ID:                          teleport.Component(teleport.ComponentWindowsDesktop),
+			CertAuthorityGetter:         muxCAGetter,
+			LocalClusterName:            clusterName,
+		})
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		go func() {
+			if err := mux.Serve(); err != nil && !utils.IsOKNetworkError(err) {
+				mux.Entry.WithError(err).Error("mux encountered err serving")
+			}
+		}()
+
+		err = srv.Serve(mux.TLS())
```

Comment:
Are we missing a `mux.Close` somewhere?

Reply:
I'm also confused why we need `mux.Serve()` and `srv.Serve()` - aren't they both accepting connections from the same underlying listener?

Reply:
> Are we missing a mux.Close somewhere?

Mux will close on `listener` being closed (which happens in `process.OnExit()`). 

> I'm also confused why we need mux.Serve() and srv.Serve() - aren't they both accepting connections from the same underlying listener?

Mux will start its own internal loop to accept/detect connections and then it pushes it upstream where `srv` can accept pushed connection in its accept loop.